# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution
</div>
****

In [11]:
import pandas as pd
import numpy as np
from scipy import stats

import matplotlib.pyplot as plt

% matplotlib inline

In [12]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [13]:
# number of callbacks for black-sounding names
sum(data[data.race=='w'].call)

235.0

In [14]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

<div class="span5 alert alert-success">
<p>Your answers to Q1 and Q2 here</p>
</div>

In [15]:
data_b = data[data.race=='b']
data_w = data[data.race=='w']
num_b = len(data_b)
num_w = len(data_w)
print("Number of observations where name is black-sounding : {}  " .format(num_b))
print("Number of observations where name is white-sounding : {}  ".format(num_w))


Number of observations where name is black-sounding : 2435  
Number of observations where name is white-sounding : 2435  


In [31]:
b_success = len(data_b[data_b.call == 1])
w_success = len(data_w[data_w.call == 1])
print(b_success, w_success)

157 235


In [32]:
p_b = 1.0 * b_success/num_b
p_w = 1.0 * w_success/num_w
print("Proportion of black sounding names getting a callback : ",p_b)
print("Proportion of white sounding names getting a callback : ",p_w)

Proportion of black sounding names getting a callback :  0.06447638603696099
Proportion of white sounding names getting a callback :  0.09650924024640657


In [34]:
print(num_b * p_b)
print(num_b * (1-p_b))
print("---")
print(num_b * p_w)
print(num_b * (1-p_w))

157.0
2278.0
---
235.0
2200.0


Validity of CLT : 

1. The names provided by the study group were evaluated by the publishers. The black sounding names were sufficiently different from the white sounding names, indicating independence.
2. The sample size of 2435 is definitely smaller than 10% of the black population
3. Selection of black or white sounding names was random, just like assignment of resume.
4. np and n(1-p) are greater than 10

Hypothesis to test

Ho : pw = pb
Ha : pw != pb

http://www.statisticshowto.com/z-test/

In [39]:
import math
z = 1.96
margin = z * math.sqrt( ( p_w*(1-p_w) / num_b) + (p_b*(1-p_b)/num_w) )

print ("Margin of error = ", margin)

Margin of error =  0.015255406349886438


In [40]:
print ("The confidence interval is given by :", p_w-p_b-z*margin,"to", p_w-p_b+z*margin)


The confidence interval is given by : 0.0021322577636681654 to 0.061933450655223


In [37]:
from statsmodels.stats.proportion import proportions_ztest as pz
pz(np.array([b_success,w_success]),np.array([num_b,num_w]),value=0)

(-4.1084121524343464, 3.9838868375850767e-05)

### Conclusion

Since the p value of our z test of proportions is less than 0.05, we have to reject the null hypothesis. This basically means that black sounding names do not get the same call back as white sounding names.